app/helpers/duties_helper.rb

```diff
@@ -42,19 +42,45 @@ def process_day_place(duties)
     result
   end
 
-  def process_duty(duties)
-    colspan = 0
+  def check_condition(span_free, span_user_id, span_request_user_id, current_duty)
+    span_free != current_duty.free || (!span_free && !current_duty.free && ((!span_request_user_id.nil? && span_request_user_id != current_duty&.request_user_id) || span_user_id != current_duty&.user&.id))
+  end
+
+  def get_index_array(duties)
+    index_array = [0]
+    span_free = duties[0].free
+    span_user_id = duties[0]&.user&.id
+    span_request_user_id = duties[0]&.request_user_id
+    (1..(duties.length - 1)).each do |index|
+      current_duty = duties[index]
+      next unless check_condition(span_free, span_user_id, span_request_user_id, current_duty)
```

Comment:
This looks a bit off...

You are using `span_free` `span_user_id` etc of the previous iteration. Probably that is intended, but that is quite some uncommon code

Reply:
If the next unless is the problem, then rubocop did that. I need to check with the previous iteration cause I check if the previous duty has the same characteristics as the current one.
Does that make sense?